### PR TITLE
fix: Break long words only in blog posts

### LINKED ISSUE #6943: Broken words on narrow viewports make reading difficult
### URL:

https://nodejs.org/en/blog/vulnerability/april-2024-security-releases#assertion-failed-in-nodehttp2http2sessionhttp2session-leads-to-http2-server-crash-cve-2024-27983---high

### Browser Name:

Firefox

### Browser Version:

129.0b7

### Operating System:

macOS 14.5

### How to reproduce the issue:

Visit a blog post containing a word that is wider than the viewport (e.g. [Wednesday, April 3, 2024 Security Releases](https://nodejs.org/en/blog/vulnerability/april-2024-security-releases#assertion-failed-in-nodehttp2http2sessionhttp2session-leads-to-http2-server-crash-cve-2024-27983---high).)

**I should see** long words broken over several lines while shorter words remain intact.

**I actually see** all words broken, making the article difficult if not impossible to read.

This issue was introduced for all viewport sizes in #6680, and partially fixed for larger viewports only in #6712. These changes break _all_ words; only long words that would cause an overflow should be broken.

| Expected | Actual |
|--------|--------|
| Expected (screen shot of article with only long words broken) | Actual (screen shot of article with all words broken) |

### PR DESCRIPTION
**Pull request by caminad**
## Description

Change `word-break: break-all;` on narrow viewports only, to `overflow-wrap: anywhere;` on all viewport sizes. `word-break: break-word;` also works, but is deprecated (see note in the [spec](https://drafts.csswg.org/css-text/#word-break-property).) I could not find a suitable Tailwind utility; [break-words](https://tailwindcss.com/docs/word-break#break-words) produces `overflow-wrap: break-word;`, which does not prevent an overflow.

## Validation

Navigate to `/en/blog/vulnerability/april-2024-security-releases#assertion-failed-in-nodehttp2http2sessionhttp2session-leads-to-http2-server-crash-cve-2024-27983---high` and check that all words are no longer broken. The page should not scroll horizontally.

| [Before (`main`)](https://nodejs.org/en/blog/vulnerability/april-2024-security-releases#assertion-failed-in-nodehttp2http2sessionhttp2session-leads-to-http2-server-crash-cve-2024-27983---high) | [After (this branch)](https://nodejs-org-git-fork-caminad-patch-1-openjs.vercel.app/en/blog/vulnerability/april-2024-security-releases#assertion-failed-in-nodehttp2http2sessionhttp2session-leads-to-http2-server-crash-cve-2024-27983---high) |
|---|---|
| Screen Shot 2024-07-24 at 14 43 27 | Screen Shot 2024-07-24 at 14 43 41 |

## Related Issues

Fixes #6943, related to #6711, related to #6721.

### Check List

- [x] I have read the [Contributing Guidelines](https://github.com/nodejs/nodejs.org/blob/main/CONTRIBUTING.md) and made commit messages that follow the guideline.
- [ ] I have run `npm run format` to ensure the code follows the style guide.
- [ ] I have run `npm run test` to check if all tests are passing.
- [ ] I have run `npx turbo build` to check if the website builds without errors.
- [ ] I've covered new added functionality with unit tests if necessary.

### CODE REVIEW

File: apps/site/layouts/layouts.module.css [hunk 1]
```diff
@@ -228,9 +228,8 @@
 
     main {
       @apply max-w-[660px]
-        gap-4
-        break-all
-        lg:break-normal;
+        gap-4;
+      overflow-wrap: anywhere;
```
Comment (bmuenzenmeyer) [inline apps/site/layouts/layouts.module.css]: small indentation fix

```suggestion
        gap-4;
        overflow-wrap: anywhere;
```

**Review by bmuenzenmeyer — APPROVED** [review]
this is a good improvement - thanks for your contribution!

File: apps/site/layouts/layouts.module.css [hunk 2]
```diff
@@ -228,9 +228,8 @@
 
     main {
       @apply max-w-[660px]
-        gap-4
-        break-all
-        lg:break-normal;
+        gap-4;
```
Comment (bmuenzenmeyer) [inline apps/site/layouts/layouts.module.css]:
```suggestion
        gap-4
```

no semicolon needed here

File: apps/site/layouts/layouts.module.css [hunk 3]
```diff
@@ -229,8 +229,7 @@
     main {
       @apply max-w-[660px]
         gap-4
-        break-all
-        lg:break-normal;
+        overflow-wrap: anywhere;
```
Comment (ovflowd) [inline apps/site/layouts/layouts.module.css]: Build is failing. Are you sure this is a valid Tailwind class name? Otherwise you need to wrap it with `[]`

Comment (bmuenzenmeyer) [reply apps/site/layouts/layouts.module.css]: I was just researching that - now realizing the original PR is mixing CSS and TW 👎

Comment (bmuenzenmeyer) [inline apps/site/layouts/layouts.module.css]: perhaps the underscore (space) is not necessary. trying this. works locally

```suggestion
        [overflow-wrap:_anywhere];
```

**Pull request merged**